docs/source/en/model_doc/chameleon.md

```diff
@@ -122,6 +122,90 @@ generate_ids = model.generate(**inputs, max_new_tokens=50)
 processor.batch_decode(generate_ids, skip_special_tokens=True, clean_up_tokenization_spaces=False)
 ```
 
+### Text to image generation
+
+Chameleon can also generate images. However, the official model checkpoint currently only supports text generation. We need to use finetuned versions such as [Anole](https://arxiv.org/abs/2407.06135) to do image generation. Here is how you can do it:
+
+```python
+from transformers import ChameleonProcessor, ChameleonForCausalLM
+import torch
+from PIL import Image
+import requests
+
+processor = ChameleonProcessor.from_pretrained("leloy/Anole-7b-v0.1-hf")
+model = ChameleonForCausalLM.from_pretrained(
+    "leloy/Anole-7b-v0.1-hf",
+    device_map="auto",
+)
+
+# Prepare a prompt
+prompt = "Generate an image of a snowman."
+
+# Preprocess the prompt
+inputs = processor(prompt, return_tensors="pt", padding=True).to(model.device)
+
+# Generate discrete image tokens
+# Note: We need to set `max_new_tokens` to 1026 since the model generates the `image_start_token` marker token first, then 1024 image tokens, and finally the `image_end_token` marker token.
+generate_ids = model.generate(
+    **inputs,
+    multimodal_generation_mode="image-only",
+    max_new_tokens=1026,
+)
```

Comment:
The current implementation ensures that we always generate 1024 image tokens (+ the start and end marker tokens). And if we can't, we just don't generate the start marker token in the first place).

As for generating two images at once... yes, that's also possible. However, if I'm not mistaken, both Chameleon & Anole were tuned to generate only one image at a time during image-only generation. But yah, the model can still generate the `boi` token instead of the `eos` token at the end of the first image block

Reply:
@zucchini-nlp I just rewrote the logits processors used for `image-only` generation mode. Can you look into it again?

Reply:
left a comment below, let's aim to have as less custom code as possible

Reply:
just removed the custom GenerationConfig!